File: 60-permutation-sequence/60-permutation-sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        global count
        count = 0
        track = []
        visited = dict()
        def permutation(n):
            global count
            if len(track) == n:
                count += 1
                if count == k:
                    return True
                return False
            for i in range(1,n+1):
                if i not in visited:
                    track.append(i)
                    visited[i] = 1
                    if permutation(n):
                        return True
                    del visited[i]
                    track.pop()
        permutation(n)
        return ''.join([str(i) for i in track])
        
            
        

File: 60-permutation-sequence/60-permutation-sequence.java
class Solution {
    public String getPermutation(int n, int k) {
        StringBuilder res = new StringBuilder();
        int num = n;
        Set<Integer> visited = new HashSet<>();
        while(num > 0){
        int curDigit = 0;
        while((curDigit + 1)*factorial(num-1) < k){
            curDigit+=1;
        }
        int count = 0;
        for(int i = 1; i < n + 1; i++){
            if(visited.contains(i))continue;
            if(curDigit == count){
                res.append(String.valueOf(i));
                visited.add(i);
            }
            count++;
        }
        k = k - curDigit*factorial(num-1);
        num--;
        }
        return res.toString();
    }
    public int factorial(int n){
        int ans  = 1;
        while(n > 0){
            ans *= n;
            n = n -1;
        }
        return ans;
        
        
        
    }

}